multinet/api.py

```diff
@@ -24,7 +27,30 @@
 @swag_from("swagger/workspaces.yaml")
 def get_workspaces() -> Any:
     """Retrieve list of workspaces."""
-    return util.stream(db.get_workspaces())
+    user = current_user()
+
+    def can_read(user: Optional[UserInfo], ws: Workspace) -> bool:
+        perms = ws["permissions"]
+
+        # No one logged in means only return the public workspaces.
+        if user is None:
+            return perms["public"]
+
+        # Otherwise, check to see if the workspace is public, or the user is at
+        # least a Reader of the workspace.
+        sub = user.sub
+        return (
+            perms["public"]
+            or perms["owner"] == sub
+            or sub in perms["readers"]
+            or sub in perms["writers"]
+            or sub in perms["maintainers"]
+        )
```

Comment:
You refactored this, but left the old function here. Looks to me like this can be removed

Reply:
Good catch. See 4f2d28d18263d845646c3cd976d467943cf7d88e.